test/suite/test_hs18.py

```diff
@@ -0,0 +1,415 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-2020 MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import time, wiredtiger, wttest, unittest
+
+def timestamp_str(t):
+    return '%x' % t
+
+# test_hs18.py
+# Test various older reader scenarios
+class test_hs18(wttest.WiredTigerTestCase):
+    conn_config = 'cache_size=5MB,eviction=(threads_max=1)'
+    session_config = 'isolation=snapshot'
+
+    def check_value(self, cursor, value):
+        cursor.set_key(str(0))
+        self.assertEqual(cursor.search(), 0)
+        self.assertEqual(cursor.get_value(), value)
+        cursor.reset()
+
+    def start_txn(self, sessions, cursors, values, i):
+        # Start a transaction that will see update 0.
+        sessions[i].begin_transaction()
+        self.check_value(cursors[i], values[i])
+
+    def test_hs18(self):
```

Comment:
May be change the function name as well?

Reply:
Yep, will do